src/policy/fees.cpp

```diff
@@ -76,116 +78,109 @@ void TxConfirmStats::UpdateMovingAverages()
     for (unsigned int j = 0; j < buckets.size(); j++)
     {
         for (unsigned int i = 0; i < confAvg.size(); i++)
+        {
             confAvg[i][j] = confAvg[i][j] * decay + curBlockConf[i][j];
+        }
         avg[j] = avg[j] * decay + curBlockVal[j];
-        txCtAvg[j] = txCtAvg[j] * decay + curBlockTxCt[j];
+        txCtAvg[j] = (txCtAvg[j] * decay) + curBlockTxCt[j];
     }
 }
 
 // returns -1 on error conditions
 double TxConfirmStats::EstimateMedianVal(int confTarget,
     double sufficientTxVal,
     double successBreakPoint,
-    bool requireGreater,
     unsigned int nBlockHeight)
 {
+    CAmount minTxFee = 0;
+    // its safe just to parse here because the validity of mintxfee was checked in init.cpp
+
+    // DEFAULT_TRANSACTION_MINFEE = 1000 defined in wallet.h, we hardcode it here as a hack fix
+    // due to --disable-wallet configs compiling policy estimator which would use wallet.h
+    // even though wallet.h isnt included when compiled with --disable-wallet
+    ParseMoney(std::to_string(GetArg("-mintxfee", 1000)), minTxFee);
+    // convert from satoshis to COIN
+    minTxFee = minTxFee / COIN;
+
     // Counters for a bucket (or range of buckets)
     double nConf = 0; // Number of tx's confirmed within the confTarget
     double totalNum = 0; // Total number of tx's that were ever confirmed
-    int extraNum = 0; // Number of tx's still in mempool for confTarget or longer
-
-    int maxbucketindex = buckets.size() - 1;
-
-    // requireGreater means we are looking for the lowest fee/priority such that all higher
-    // values pass, so we start at maxbucketindex (highest fee) and look at successively
-    // smaller buckets until we reach failure.  Otherwise, we are looking for the highest
-    // fee/priority such that all lower values fail, and we go in the opposite direction.
-    unsigned int startbucket = requireGreater ? maxbucketindex : 0;
-    int step = requireGreater ? -1 : 1;
-
-    // We'll combine buckets until we have enough samples.
-    // The near and far variables will define the range we've combined
-    // The best variables are the last range we saw which still had a high
-    // enough confirmation rate to count as success.
-    // The cur variables are the current range we're counting.
-    unsigned int curNearBucket = startbucket;
-    unsigned int bestNearBucket = startbucket;
-    unsigned int curFarBucket = 0;
-    unsigned int bestFarBucket = startbucket;
-
-    bool foundAnswer = false;
-    unsigned int bins = unconfTxs.size();
-
-    // Start counting from highest(default) or lowest fee/pri transactions
-    for (int bucket = startbucket; bucket >= 0 && bucket <= maxbucketindex; bucket += step)
+    int32_t extraNum = 0; // Number of tx's still in mempool for confTarget or longer
+
+    int32_t maxbucketindex = buckets.size() - 1;
```

Comment:
If we let the machine choose its preferred integer size the code will execute faster and be smaller, since the "native" size computes in 1 instruction.  I think that we can assume our target machine's word size is 32bits or bigger...